include/eosio/to_json.hpp

```diff
@@ -211,119 +195,78 @@ result<void> to_json(const std::variant<T...>& obj, S& stream) {
    };
 
 template <typename T, typename S>
-result<void> to_json(const T& t, S& stream) {
-   result<void> ok    = outcome::success();
+void to_json(const T& t, S& stream) {
    bool         first = true;
-   OUTCOME_TRY(stream.write('{'));
+   stream.write('{');
    eosio::for_each_field<T>([&](const char* name, auto&& member) {
-      if (ok) {
-          auto addfield = [&]() {
-            if (first) {
-               auto r = increase_indent(stream);
-               if (!r) {
-                  ok = r;
-                  return;
-               }
-               first = false;
-            } else {
-               auto r = stream.write(',');
-               if (!r) {
-                  ok = r;
-                  return;
-               }
-            }
-            auto r = write_newline(stream);
-            if (!r) {
-               ok = r;
-               return;
-            }
-            r = to_json(name, stream);
-            if (!r) {
-               ok = r;
-               return;
-            }
-            r = write_colon(stream);
-            if (!r) {
-               ok = r;
-               return;
-            }
-            r = to_json(member(&t), stream);
-            if (!r) {
-               ok = r;
-               return;
-            }
-         };
-
-         auto m = member(&t);
-         using member_type = std::decay_t<decltype(m)>;
-         if constexpr ( not is_std_optional<member_type>::value ) {
-            addfield();
+       auto addfield = [&]() {
+         if (first) {
+            increase_indent(stream);
+            first = false;
          } else {
-            if( !!m ) 
-               addfield();
+            stream.write(',');
          }
+         write_newline(stream);
+         to_json(name, stream);
+         write_colon(stream);
+         to_json(member(&t), stream);
+      };
+
+      auto m = member(&t);
+      using member_type = std::decay_t<decltype(m)>;
+      if constexpr ( not is_std_optional<member_type>::value ) {
+         addfield();
+      } else {
+         if( !!m || true )
```

Comment:
Why was the `|| true` added?

Reply:
The tests fail otherwise.

Reply:
Specifically, this change to the format is not being handled consistently in other parts of abieos.

Reply:
ok, maybe add a TODO to fix this.